### ISSUE TITLE
Allow node-role labeling

### ISSUE THREAD

**Issue by paulbsch** [weaveworks/eksctl#580]
**Why do you want this feature?**
In our use case, we utilize node-role.kuberneties.io/{role-name} labels.  It is also useful for identifying nodes in the cluster when using tools like kubectl.  Example:
```
$ kubectl get nodes
NAME                                           STATUS    ROLES             AGE       VERSION
ip-192-168-66-184.us-west-2.compute.internal   Ready     jarvice-compute   37m       v1.11.5
ip-192-168-71-24.us-west-2.compute.internal    Ready     jarvice-compute   37m       v1.11.5
ip-192-168-75-4.us-west-2.compute.internal     Ready     jarvice-system    37m       v1.11.5
```

Currently, eksctl doesn't allow any label matching the kubernetes.io domain.  This makes perfect sense with respect to labels that should be immutable ("kubernetes.io/arch", kubernetes.io/hostname", etc.)  However, node role labels don't seem to fall into that category.

**What feature/behavior/change do you want?**
Please alllow node role labeling.  i.e. node-role.kubernetes.io/{role-name}  e.g. node-role.kubernetes.io/compute

**Comment by errordeveloper** [comment]
Do we currently return `unknown 'kubernetes.io' or 'k8s.io' labels were specified` kind of error?

It should be fairly easy to add `node-role.kubernetes.io`, would you like to try adding it yourself?

I believe it just needs to be added to the list of domain prefixes here:

https://github.com/weaveworks/eksctl/blob/2319fa8dd39c755bc2e296dff3bc36595da0b560/pkg/apis/eksctl.io/v1alpha4/validation.go#L68-L72

**Comment by paulbsch** [comment]
It does return that error, yes.

Adding that domain prefix solves the issue.  I will submit a pull request.

**Comment by errordeveloper** [comment]
Closed via #582.
